sql-cli/pyproject.toml

```diff
@@ -38,10 +38,12 @@ networkx = [
 ]
 Jinja2 = ">=2.10.1, <=3.1"
 python-frontmatter = "^1.0.0"
-PyYAML = ">5.4.1"
-apache-airflow = ">2.0"
+PyYAML = ">=5.4.1"
+# the Python SDK 1.2 enforces pandas >= 1.3.4,
+# but Airflow 2.1 has constraints pandas == 1.3.3
 python-dotenv = "^0.21.0"
 astro-sdk-python = {extras = ["all"], version = "^1.2.0"}
+apache-airflow = ">2.1"
```

Comment:
https://github.com/astronomer/astro-sdk/actions/runs/3450993943/jobs/5759928610

When we attempt to use the Airflow constraints for 2.1, these are the conflicts:
```
The conflict is caused by the following:
    apache-airflow 2.1.4 depends on pandas<2.0 and >=0.17.1; python_version >= "3.7"
    astro-sdk-python[all] 1.2.1 depends on pandas<2.0.0 and >=1.3.4
    apache-airflow 2.1.4 depends on pandas<2.0 and >=0.17.1; python_version >= "3.7"
    astro-sdk-python[all] 1.2.0 depends on pandas<2.0.0 and >=1.3.4
    The user requested (constraint) pandas==1.3.3
```

If we ignore the constraints for 2.1, there are a couple of errors:
```
tests/test___main__.py ........FF......F.F..FF..FF.........FFF...                      [ 60%]
tests/test_configuration.py ..                                                         [ 62%]
tests/test_connections.py FF                                                           [ 65%]
tests/test_dag_generator.py .......                                                    [ 75%]
tests/test_project.py ..                                                               [ 78%]
tests/test_run_dag.py ...ss...                                                         [ 90%]
tests/test_sql_directory_parser.py .......                                             [100%]
```

I consider it outside the scope of the current ticket to support 2.1.  I logged a ticket for when we decide to prioritise this:
https://github.com/astronomer/astro-sdk/issues/1240